rct229/rulesets/ashrae9012019/section19/section19rule30.py

```diff
@@ -0,0 +1,63 @@
+from rct229.rule_engine.partial_rule_definition import PartialRuleDefinition
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.rulesets.ashrae9012019.ruleset_functions.baseline_system_type_compare import (
+    baseline_system_type_compare,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.baseline_systems.baseline_system_util import (
+    HVAC_SYS,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_baseline_system_types import (
+    get_baseline_system_types,
+)
+
+APPLICABLE_SYS_TYPES = [HVAC_SYS.SYS_6, HVAC_SYS.SYS_8]
+
+
+class Section19Rule30(RuleDefinitionListIndexedBase):
+    """Rule 30 of ASHRAE 90.1-2019 Appendix G Section 19 (Envelope)"""
+
+    def __init__(self):
+        super(Section19Rule30, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section19Rule30.HVACRule(),
+            index_rmr="baseline",
+            id="19-30",
+            description="or Systems 6 and 8, only the terminal-unit fan and reheat coil shall be energized to meet heating set point during unoccupied hours in the baseline design.",
+            ruleset_section_title="HVAC - General",
+            standard_section="Exception to Section G3.1.2.4",
+            is_primary_rule=False,
+            rmr_context="ruleset_model_descriptions/0",
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+        )
+
+    def create_data(self, context, data):
+        rmd_b = context.baseline
+
+        return {"baseline_system_types_dict_b": get_baseline_system_types(rmd_b)}
+
+    class HVACRule(PartialRuleDefinition):
+        def __init__(self):
+            super(Section19Rule30.HVACRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+            )
+
+        def applicability_check(self, context, calc_vals, data):
+            rmd_b = context.baseline
+            baseline_system_types_dict_b = get_baseline_system_types(rmd_b)
+
```

Comment:
I think this has the same issue as the #1165 
In HVAC level, you do not need to loop a list to get the result. This is specific in an HVAC, just need to check whether this particular HVAC applicability.